storage/s3.go

```diff
@@ -125,12 +133,19 @@ func (s *S3) Stat(ctx context.Context, url *url.URL) (*Object, error) {
 
 	etag := aws.StringValue(output.ETag)
 	mod := aws.TimeValue(output.LastModified)
-	return &Object{
+
+	obj := &Object{
 		URL:     url,
 		Etag:    strings.Trim(etag, `"`),
 		ModTime: &mod,
 		Size:    aws.Int64Value(output.ContentLength),
-	}, nil
+	}
+
+	if res, ok := output.Metadata[metadataKeyRetryCode]; ok && s.noSuchUploadRetryCount > 0 {
+		obj.retryCode = *res
+	}
```

Comment:
I think something like below would make it more readable. One may easily spot this only works with `noSuckUploadRetryCode > 0`
```suggestion
	if s.noSuchUploadRetryCount > 0 {
		if retryCode, ok := output.Metadata[metadataKeyRetryCode];ok {
			obj.retryCode = *retryCode
		}
	}
```